src/services/shortLink/service.ts

```diff
@@ -0,0 +1,87 @@
+import { PermissionLevel } from '@graasp/sdk';
+
+import { UnauthorizedMember } from '../../utils/errors';
+import { Repositories } from '../../utils/repositories';
+import { validatePermission } from '../authorization';
+import { Actor } from '../member/entities/member';
+import { ShortLink } from './entities/ShortLink';
+import { getPlatformHost, getPlatformPrefix, getUrlHost } from './utils';
+
+export class ShortLinkService {
+  async post(member: Actor, shortLink: ShortLink, repositories: Repositories) {
+    const { shortLinkRepository, itemRepository } = repositories;
+
+    if (!member) throw new UnauthorizedMember();
+    const item = await itemRepository.get(shortLink.item_id);
+
+    // check that the member can admin the item to be allowed to create short link
+    await validatePermission(repositories, PermissionLevel.Admin, member, item);
+
+    return shortLinkRepository.createOne(shortLink);
+  }
+
+  async get(alias: string, repositories: Repositories) {
+    const { shortLinkRepository } = repositories;
+    const shortLink = await shortLinkRepository.get(alias);
+
+    return shortLink;
+  }
+
+  async getItem(member: Actor, itemId: string, repositories: Repositories) {
+    const { shortLinkRepository, itemRepository } = repositories;
+
+    if (!member) throw new UnauthorizedMember();
+    const item = await itemRepository.get(itemId);
+
+    // check that the member can admin the item to be allowed to create short link
+    await validatePermission(repositories, PermissionLevel.Admin, member, item);
+
+    return await shortLinkRepository.getItem(itemId);
+  }
+
+  async getRealPath(alias: string, repositories: Repositories) {
+    const shortLink = await this.get(alias, repositories);
+    const host = getPlatformHost(shortLink.platform);
+    const prefix = getPlatformPrefix(shortLink.platform);
+    return {
+      url: `${getUrlHost(host)}${prefix}/${shortLink.item_id}`,
+      platform: shortLink.platform,
+      alias: shortLink.alias,
+      item_id: shortLink.item_id,
+    };
+  }
+
+  async delete(member: Actor, alias: string, repositories: Repositories) {
+    const { shortLinkRepository, itemRepository } = repositories;
+
+    if (!member) throw new UnauthorizedMember();
+    const shortLink = await shortLinkRepository.get(alias);
+    const item = await itemRepository.get(shortLink.item_id);
```

Comment:
You can get the `item` by using the `relations: { item: true }` in your typeorm `this.find()` query.